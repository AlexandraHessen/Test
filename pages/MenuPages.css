.MenuPages{
    font-size: 1.25em;
    font-weight: bold;
    list-style-type: none;

}
.MenuPages .PageLink{
    text-decoration: none;
    padding: 10px;
    color: #fff
}

.ActivePageLink {
    background: #57ad26;;
    border-radius: 5px;

}

@media all and (max-width: 500px){
    .MenuPages .PageLink{
        padding: 5px;
    }

    .MenuPages{
        font-size: 1.15em;
    }
}