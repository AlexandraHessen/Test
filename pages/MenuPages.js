"use strict";
import React from "react";
import { NavLink } from 'react-router-dom';

import "./MenuPages.css"

class MenuPages extends React.Component{
      // нельзя pure componenet
    render(){
        return(
            <ul className="MenuPages">
                <li><NavLink to="/" exact className="PageLink" activeClassName="ActivePageLink">Главная</NavLink></li>
                <li><NavLink to="/catalog/1" className="PageLink" activeClassName="ActivePageLink">Каталог</NavLink></li>
                <li><NavLink to="/contacts" className="PageLink" activeClassName="ActivePageLink">Контакты</NavLink></li>
                <li><NavLink to="/basket" className="PageLink"activeClassName="ActivePageLink">Корзина</NavLink></li>
            </ul>
        )
    }

}

export default MenuPages;