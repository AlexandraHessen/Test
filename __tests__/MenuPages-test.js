"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import MenuPages from '../pages/MenuPages';
                    //импортируем файл который будем тестировать

test('MenuPages', () => {
                            // создаём тестовую версию компонента который будем тестировать
  const component = renderer.create(
    <BrowserRouter>
      <MenuPages />
    </BrowserRouter>
  );

                            // получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
