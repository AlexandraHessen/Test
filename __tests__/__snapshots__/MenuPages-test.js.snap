// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuPages 1`] = `
<ul
  className="MenuPages"
>
  <li>
    <a
      aria-current={false}
      className="PageLink"
      href="/"
      onClick={[Function]}
      style={undefined}
    >
      Главная
    </a>
  </li>
  <li>
    <a
      aria-current={false}
      className="PageLink"
      href="/catalog/1"
      onClick={[Function]}
      style={undefined}
    >
      Каталог
    </a>
  </li>
  <li>
    <a
      aria-current={false}
      className="PageLink"
      href="/contacts"
      onClick={[Function]}
      style={undefined}
    >
      Контакты
    </a>
  </li>
  <li>
    <a
      aria-current={false}
      className="PageLink"
      href="/basket"
      onClick={[Function]}
      style={undefined}
    >
      Корзина
    </a>
  </li>
</ul>
`;
